File: go/cyphr/transaction.go
package cyphr

// Transaction is a sequence of related mutation cozies.
// In Cyphr, mutation cozies MUST be grouped by transaction.
type Transaction []*ParsedCoz

// CommitTransaction is a specialized sequence of cozies containing the finality marker.
// The commit transaction MUST be the exact last transaction in the commit.
type CommitTransaction []*ParsedCoz

File: rs/cyphr/src/transaction.rs
use crate::parsed_coz::VerifiedCoz;

/// A sequence of one or more related mutation cozies.
/// In Cyphr, mutation cozies MUST be grouped by transaction.
#[derive(Debug, Clone)]
pub struct Transaction(pub Vec<VerifiedCoz>);

impl Transaction {
    /// Returns true if this transaction contains a commit finalizer cozy.
    pub fn is_commit(&self) -> bool {
        self.0.iter().any(|cz| cz.arrow().is_some())
    }
}
